test/e2e/ibc_transfer_e2e_test.go

```diff
@@ -39,16 +40,114 @@ func (s *IBCTransferTestSuite) TearDownSuite() {
 	}
 }
 
+func getFirstIBCDenom(balance sdk.Coins) string {
+	// Look up the ugly IBC denom
+	denoms := balance.Denoms()
+	var denomB string
+	for _, d := range denoms {
+		if strings.HasPrefix(d, "ibc/") {
+			denomB = d
+			break
+		}
+	}
+	return denomB
+}
+
 func (s *IBCTransferTestSuite) Test1IBCTransfer() {
-	bbnChain := s.configurer.GetChainConfig(0)
+	denom := "ubbn"
+	amount := int64(1_000_000)
+	delta := float64(10000) // Tolerance to account for gas fees
+
+	transferCoin := sdk.NewInt64Coin(denom, amount)
+
+	bbnChainA := s.configurer.GetChainConfig(0)
+	bbnChainB := s.configurer.GetChainConfig(1)
+
+	babylonNodeA, err := bbnChainA.GetNodeAtIndex(2)
+	s.NoError(err)
+	babylonNodeB, err := bbnChainB.GetNodeAtIndex(2)
+	s.NoError(err)
+
+	val := initialization.ValidatorWalletName
+
+	// Check balance of val in chain-A (Node 3)
+	addrA := babylonNodeA.GetWallet(val)
+	balanceA, err := babylonNodeA.QueryBalances(addrA)
+	s.Require().NoError(err)
+	// Confirm val on A has enough funds
+	s.Assert().GreaterOrEqual(balanceA.AmountOf(denom).Int64(), amount)
+
+	addrB := babylonNodeB.GetWallet(val)
+	balanceB, err := babylonNodeB.QueryBalances(addrB)
+	s.Require().NoError(err)
+	// Only one denom in B
+	s.Require().Len(balanceB, 1)
+
+	// Send transfer from val in chain-A (Node 3) to val in chain-B (Node 3)
+	babylonNodeA.SendIBCTransfer(val, addrB, "transfer", transferCoin)
+
+	time.Sleep(10 * time.Second)
+
+	// Check that the transfer is successful.
+	// Amounts have been discounted from val in chain-A and added (as a wrapped denom) to val in chain-B
+	balanceA2, err := babylonNodeA.QueryBalances(addrA)
+	s.Require().NoError(err)
+	s.Assert().InDelta(balanceA.Sub(transferCoin).AmountOf(denom).Int64(), balanceA2.AmountOf(denom).Int64(), delta)
```

Comment:
small nit, can we use `require.eventually` instead of timeout

Reply:
Yes, can do, but not sure what to wait for. Will take a look. 10 seconds is not so bad... it was 30 before. :-)

Reply:
Ah, `eventually` is just the condition and a timeout, good.